src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -369,11 +369,17 @@ protected override async Task DeleteMessagesToAsync(string persistenceId, long t
 
         private JournalEntry ToJournalEntry(IPersistentRepresentation message)
         {
-            object payload = message.Payload;
+            List<string> tags; 
```

Comment:
Change this to an `string[]` and use `Array.Empty<string>` to prevent allocations where not needed.

Reply:
Done